### PR TITLE
Simplify BPF tracing implementation

### PR DESCRIPTION
**Pull request by chenhengqi**
* Remove all `__always_inline` from BPF program, this reduces size of `trace_bpfel_x86.o` file
* Replace `memcpy`(from `<string.h>`) with `__builtin_memcpy`
* Simplify `get_goroutine_id` implementation, we don't need a extra BPF ringbuf

### CODE REVIEW

**Comment by derekparker** [comment]
These changes look nice but pulling your branch and running the EBPF test in `cmd/dlv` fails.

**Review by derekparker — CHANGES_REQUESTED** [review]
These changes look nice but pulling your branch and running the EBPF test in cmd/dlv fails.

**Comment by chenhengqi** [comment]
Will investigate.

File: pkg/proc/internal/ebpf/bpf/trace.bpf.c [hunk 1]
```diff
@@ -244,4 +213,4 @@ int uprobe__dlv_trace(struct pt_regs *ctx) {
     return 0;
 }
 
-char _license[] SEC("license") = "GPL";
\ No newline at end of file
+char _license[] SEC("license") = "GPL";
```
Comment (aarzilli) [inline pkg/proc/internal/ebpf/bpf/trace.bpf.c:216]: Unrelated to this PR but I didn't notice it before. Should this really say GPL?

Comment (chenhengqi) [reply pkg/proc/internal/ebpf/bpf/trace.bpf.c:216]: `bpf_probe_read_user` is used in this BPF program, and this helper is GPL only.

https://github.com/torvalds/linux/blob/8bb7eca972ad531c9b149c0a51ab43a417385813/kernel/trace/bpf_trace.c#L169-L176

Please also refer to https://github.com/torvalds/linux/blob/master/Documentation/bpf/bpf_licensing.rst.

Comment (aarzilli) [reply pkg/proc/internal/ebpf/bpf/trace.bpf.c:216]: That says that to call GPL only functions the probe needs to be GPL compatible, which many licenses are.

Comment (chenhengqi) [reply pkg/proc/internal/ebpf/bpf/trace.bpf.c:216]: Right, but kernel only accept certain combinations.

https://github.com/torvalds/linux/blob/v5.15/include/linux/license.h#L5-L13

Comment (derekparker) [reply pkg/proc/internal/ebpf/bpf/trace.bpf.c:216]: We should rewrite it to be `Dual MIT/GPL`.

**Comment by derekparker** [comment]
Closing as there have been no recent updates.

**Pull request closed without merging**